app/controllers/api/service_templates_controller.rb

```diff
@@ -4,5 +4,63 @@ class ServiceTemplatesController < BaseController
     include Subcollections::Tags
     include Subcollections::ResourceActions
     include Subcollections::ServiceRequests
+
+    def create_resource(_type, _id, data)
+      # Temporarily only supporting atomic.
+      raise 'Service Type composite not supported' if data['service_type'] == 'composite'
+      create_atomic(data).tap(&:save!)
+    rescue => err
+      raise BadRequestError, "Could not create Service Template - #{err}"
+    end
+
+    private
+
+    def create_atomic(data)
+      service_template = ServiceTemplate.new(data.except('request_info'))
+      dialog = nil
+      if data.key?('request_info')
+        service_template.add_resource(create_service_template_request(data['request_info']))
```

Comment:
@jntullo Adding resources to a service_template, you can follow the examples in catalog_controller. Basically there is no common code. You need to call generic type specific methods to add them: https://github.com/ManageIQ/manageiq/blob/master/app/controllers/catalog_controller.rb#L1507 https://github.com/ManageIQ/manageiq/blob/master/app/controllers/catalog_controller.rb#L1509 https://github.com/ManageIQ/manageiq/blob/master/app/controllers/catalog_controller.rb#L1514